zoom-pan-options.js

```diff
@@ -68,88 +81,97 @@ function _constrainView_Override({x, y, scale}) {
-    canvas.pan({scale: dz * canvas.stage.scale.x});
+  const multiplier = getSetting('zoom-speed-multiplier');
+  let dz = -event.deltaY * 0.0005 * multiplier + 1;
+
+  if (!getSetting('zoom-around-cursor')) {
+    canvas.pan({ scale: dz * canvas.stage.scale.x });
     return;
   }
+
   const scale = dz * canvas.stage.scale.x;
   const d = canvas.dimensions;
   const max = CONFIG.Canvas.maxZoom;
   const min = 1 / Math.max(d.width / window.innerWidth, d.height / window.innerHeight, max);
+
   if (scale > max || scale < min) {
-    canvas.pan({scale: scale > max ? max : min});
-    console.log(`Zoom/Pan Options | scale limit reached (${scale}).`)
-    return
+    canvas.pan({ scale: scale > max ? max : min });
+    console.log('zoom-pan-options |', `scale limit reached (${scale}).`);
+    return;
   }
+
   // Acquire the cursor position transformed to Canvas coordinates
   const t = canvas.stage.worldTransform;
   const dx = ((-t.tx + event.clientX) / canvas.stage.scale.x - canvas.stage.pivot.x) * (dz - 1);
   const dy = ((-t.ty + event.clientY) / canvas.stage.scale.y - canvas.stage.pivot.y) * (dz - 1);
   const x = canvas.stage.pivot.x + dx;
   const y = canvas.stage.pivot.y + dy;
-  canvas.pan({x, y, scale});
+  canvas.pan({ x, y, scale });
 }
 
 function panWithTouchpad(event) {
-  const multiplier = 1 / canvas.stage.scale.x * getSetting('pan-speed-multiplier')
-  const x = canvas.stage.pivot.x + event.deltaX * multiplier
-  const y = canvas.stage.pivot.y + event.deltaY * multiplier
-  canvas.pan({x, y})
+  const multiplier = (1 / canvas.stage.scale.x) * getSetting('pan-speed-multiplier');
+  const x = canvas.stage.pivot.x + event.deltaX * multiplier;
+  const y = canvas.stage.pivot.y + event.deltaY * multiplier;
+  canvas.pan({ x, y });
 }
 
-Hooks.on("init", function () {
-  game.settings.register("zoom-pan-options", "zoom-around-cursor", {
-    name: "Zoom around cursor",
-    hint: "Center zooming around cursor. Does not apply to zooming with pageup or pagedown.",
-    scope: "client",
+Hooks.on('init', function () {
+  game.settings.register('zoom-pan-options', 'zoom-around-cursor', {
+    name: 'Zoom around cursor',
+    hint: 'Center zooming around cursor. Does not apply to zooming with pageup or pagedown.',
+    scope: 'client',
     config: true,
     default: true,
-    type: Boolean
-  })
-  game.settings.register("zoom-pan-options", "touchpad-scroll", {
-    name: "Zoom by pinching, pan by dragging (Touchpad mode)",
-    hint: "Pan with two-finger drag (or vertical/horizontal scroll)." +
-      " Zoom with two-finger pinch (or Ctrl+scroll)." +
-      " Precisely rotate a token with Ctrl+Shift+scroll.",
-    scope: "client",
+    type: Boolean,
+  });
+  game.settings.register('zoom-pan-options', 'touchpad-scroll', {
+    name: 'Touchpad/Scrollwheel Mode',
+    hint:
+      'Pan with two-finger drag (or scroll wheel, Shift + scroll for Horizontal). Zoom with two-finger pinch (or Ctrl + scroll).',
```

Comment:
I noticed you removed "Precisely rotate a token with Ctrl+Shift+scroll."

Please add the new behavior - defining exactly how it is with alt, shift, ctrl, etc. at least how it's different from default behavior.

Reply:
The Precision rotate occurs while only holding Alt + Scrolling in touchpad mode with these changes.